src/proc/constants.rs

```diff
@@ -253,6 +261,102 @@ impl<'a> ConstantSolver<'a> {
             inner,
         }))
     }
+
+    fn binary_op(
+        &mut self,
+        op: BinaryOperator,
+        left: Handle<Constant>,
+        right: Handle<Constant>,
+    ) -> Result<Handle<Constant>, ConstantSolvingError> {
+        use std::mem::discriminant;
+
+        let left = &self.constants[left].inner;
+        let right = &self.constants[right].inner;
+
+        let inner = match (left, right) {
+            (
+                ConstantInner::Scalar {
+                    value: left_value,
+                    width,
+                },
+                ConstantInner::Scalar {
+                    value: right_value, ..
+                },
+            ) if discriminant(left_value) == discriminant(right_value) => {
+                let value = match op {
+                    BinaryOperator::Equal => ScalarValue::Bool(left_value == right_value),
+                    BinaryOperator::NotEqual => ScalarValue::Bool(left_value != right_value),
+
+                    _ => match (left_value, right_value) {
+                        (ScalarValue::Sint(a), ScalarValue::Sint(b)) => match op {
```

Comment:
let's just wrap it up, i.e. `=> ScalarValue::Sint(match op {...})`
can be applied to other places too

Reply:
This isn't possible because in the comparison operations we return a boolean

Reply:
If we are talking about scalars only, the booleans could be moved outside and processed together with `Equal`/`NotEqual`.